src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -172,10 +201,15 @@ async def pylint_lint(
         sorted(field_set.address.reference() for field_set in field_sets)
     )
 
-    process = requirements_pex.create_process(
+    process = pylint_runner_pex.create_process(
         python_setup=python_setup,
         subprocess_encoding_environment=subprocess_encoding_environment,
-        pex_path=f"./pylint_runner.pex",
+        pex_path="./pylint_runner.pex",
+        # NB: Pylint source plugins must be explicitly loaded via PYTHONPATH. The value must
+        # point to the plugin's directory, rather than to a parent directory, because
+        # `load-plugins` takes a module name rather than a path to the module. This means users
+        # must specify the parent directory as a source root pattern.
+        env={"PYTHONPATH": "./"} if pylint.source_plugins else None,
```

Comment:
Rather than adding this at the root, it would be good to nest it under something (which would avoid pylint inspecting unrelated chroot content, which could potentially collide): so materialize all of the plugin dependencies at something like `_plugins/`, and then load via `"PYTHONPATH": "_plugins/"`.